### PR TITLE
feat(devtools): remove base lib and adopt footstone

### PR DESCRIPTION
**Pull request by xiaoguo0742**
# Pre-PR Checklist

- [x] I added/updated relevant documentation.
- [x] I followed the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters to submit commit message.
- [x] I squashed the repeated code commits.
- [x] I signed the [CLA].
- [x] I added/updated test cases to check the change I am making.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: devtools/devtools-backend/include/tunnel/ws/web_socket_channel.h [hunk 1]
```diff
@@ -76,7 +76,7 @@ namespace detail {
  */
 template<typename Exception>
 inline void throw_exception(const Exception &e) {
-  BACKEND_LOGE(TDF_BACKEND, " asio exception:%s", e.what());
+  FOOTSTONE_DLOG(ERROR) << " asio exception:%s" << e.what();
```
Comment (churchill-zhang) [inline devtools/devtools-backend/include/tunnel/ws/web_socket_channel.h:79]: %s 不用了吧

File: devtools/devtools-backend/src/module/domain_dispatch.cc [hunk 2]
```diff
@@ -139,10 +141,10 @@ bool DomainDispatch::ReceiveDataFromFrontend(const std::string& data_string) {
 
 void DomainDispatch::DispatchToVm(const std::string& data) {
 #if defined(JS_V8) && !defined(V8_WITHOUT_INSPECTOR)
-  BACKEND_LOGD(TDF_BACKEND, "JSDebugger, params=%s.", data.c_str());
+  FOOTSTONE_DLOG(INFO) << "JSDebugger, params=%s." << data.c_str();
```
Comment (churchill-zhang) [inline devtools/devtools-backend/src/module/domain_dispatch.cc:144]: 同上

File: devtools/devtools-backend/src/module/model/css_model.cc [hunk 3]
```diff
@@ -111,7 +111,7 @@ nlohmann::json CssModel::UpdateDomTreeAndGetStyleTextJson(const nlohmann::json&
   if (dom_tree_adapter) {
     auto update_node_id = node_id_;
     dom_tree_adapter->UpdateDomTree(result_metas, [update_node_id](const bool is_success) {
-      BACKEND_LOGI(TDF_BACKEND, "CSS, update dom tree, id: %ld, success: %d", update_node_id, is_success);
+      FOOTSTONE_DLOG(INFO) << "CSS, update dom tree, id: %ld, success: %d" << update_node_id << is_success;
```
Comment (churchill-zhang) [inline devtools/devtools-backend/src/module/model/css_model.cc:114]: 同上

File: modules/footstone/include/footstone/string_utils.h [hunk 4]
```diff
@@ -23,58 +23,59 @@
...
   /**
    * @brief simple characterization method
    * @tparam T
    * @param value object that needs to be characterized
    * @return
    */
-  template <typename T>
+  template<typename T>
   static std::string ToString(T value) {
     std::stringstream temp_stream;
```
Comment (churchill-zhang) [inline modules/footstone/include/footstone/string_utils.h:74]: 貌似没考虑异常？

File: devtools/devtools-backend/include/tunnel/ws/web_socket_channel.h (hunk 1 above)
Comment (lavnFan) [reply devtools/devtools-backend/include/tunnel/ws/web_socket_channel.h:79]: 已改

File: devtools/devtools-backend/src/module/domain_dispatch.cc (hunk 2 above)
Comment (lavnFan) [reply devtools/devtools-backend/src/module/domain_dispatch.cc:144]: 已改

File: devtools/devtools-backend/src/module/model/css_model.cc (hunk 3 above)
Comment (lavnFan) [reply devtools/devtools-backend/src/module/model/css_model.cc:114]: 已改

File: modules/footstone/include/footstone/string_utils.h (hunk 4 above)
Comment (lavnFan) [reply modules/footstone/include/footstone/string_utils.h:74]: Strong guarantee: if an exception is thrown, there are no side effects.
异常安全的

**Review by churchill-zhang — APPROVED** [review]

**Review by zoomchan-cxj — APPROVED** [review]
LGTM

**Pull request merged**